homeassistant/components/tile/__init__.py

```diff
@@ -63,18 +50,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     await gather_with_limited_concurrency(
         DEFAULT_INIT_TASK_LIMIT, *coordinator_init_tasks
     )
-    hass.data.setdefault(DOMAIN, {})
-    hass.data[DOMAIN][entry.entry_id] = TileData(coordinators=coordinators, tiles=tiles)
+    entry.runtime_data = coordinators
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: TileConfigEntry) -> bool:
     """Unload a Tile config entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
-        hass.data[DOMAIN].pop(entry.entry_id)
-
-    return unload_ok
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
```

Comment:
Does `entry_data` allow this to be removed?

Reply:
That happens automatically :D